python/dlr/counter/model_metric.py

```diff
@@ -0,0 +1,66 @@
+import concurrent.futures
+import logging
+import time
+import hashlib
+import json
+
+from .utils import resturlutils
+from . import config
+from .model_exec_counter import ModelExecCounter
+
+
+class ModelMetric(object):
+    _pub_model_metric = True
+    _instance = None
+    MODEL_RUN = 3
+
+    @staticmethod
+    def get_instance(uuid):
+        """return single instance of class"""
+        if ModelMetric._instance is None:
+            ModelMetric._instance = ModelMetric(uuid)
+        return ModelMetric._instance
+
+    def __init__(self, uuid=0):
+        try:
+            self.client = resturlutils.RestUrlUtils()
+            self.uuid = uuid
+            # start loop
+            executor = concurrent.futures.ThreadPoolExecutor(max_workers=config.CALL_HOME_MAX_WORKERS_THREADS)
+            executor.submit(self.push_model_metric)
+            logging.info("model metric thread pool execution started")
+        except Exception as e:
+            logging.exception("model metric thread pool not started", exc_info=True)
+
+    def push_model_metric(self):
+        """publishing model run metric"""
+        while ModelMetric._pub_model_metric:
+            time.sleep(config.CALL_HOME_MODEL_RUN_COUNT_TIME_SECS)
+            for key, val in ModelExecCounter.get_model_counts_dict().items():
+                self.model_run_info_publish(ModelMetric.MODEL_RUN, key, val)
+            ModelExecCounter.clear_model_counts()
+
+    def model_run_info_publish(self, model_event_type, model, count=0):
+        """push model load information at time model load time"""
+        try:
+            _md5model = hashlib.md5(model.encode())
```

Comment:
I'd like this to be a separate utility function. `def hash_model_name()`

Reply:
+1

Reply:
Already Done